### PR TITLE
replace collections extension with commons extension

### PR DESCRIPTION
**Pull request by matthewhanson**
Changes the Collections extension with the Commons extension. 

This allows putting properties that belong to an `Item` into it's `Collection` and have it apply to all member `Item`s.

### CODE REVIEW

**Comment by m-mohr** [comment]
Not reviewed yet, but I'd like to have all extensions in folders. Therefore I'D suggest moving the file in a commons folder. Same. Should be done with the EO extension, but that can be a separate PR of course.

File: extensions/commons/README.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
+# Commons Extension Spec (`c`)
```
Comment (m-mohr) [inline extensions/commons/README.md]: Why is there the prefix 'c' mentioned? The extension page says there's no prefix.

**Comment by matthewhanson** [comment]
@m-mohr I did move it into a 'commons' folder.

File: extensions/commons/README.md (hunk 1 above)
Comment (matthewhanson) [reply extensions/commons/README.md]: ah, thanks, fixing...

**Comment by m-mohr** [comment]
Thanks for moving it! I resolved the merge conflicts with another PR and added minor improvements.

**Comment by matthewhanson** [comment]
@m-mohr examples added and README updated

**Pull request merged**